onnx/test/symbolic_shape_test.py

```diff
@@ -35,27 +35,34 @@ def _assert_valueInfo_shape(self, onnx_model, valueInfo):
                     else:
                         assert dim.dim_value == nameShape[v.name][dim_i], '%s' % (onnx_model)
 
-    def _count_unqiue_dim_param_number(self, onnx_model):
+    def _count_unqiue_dim_param_number(self, onnx_model):  # type: (ModelProto) -> int
         symbol_shape_set = set()
-        for input in onnx_model.graph.input:
-            for dim_i in range(len(input.type.tensor_type.shape.dim)):
-                dim = input.type.tensor_type.shape.dim[dim_i]
-                if dim.dim_param:
-                    symbol_shape_set.add(dim.dim_param)
-        for output in onnx_model.graph.output:
-            for dim_i in range(len(output.type.tensor_type.shape.dim)):
-                dim = output.type.tensor_type.shape.dim[dim_i]
-                if dim.dim_param:
-                    symbol_shape_set.add(dim.dim_param)
-        for v in onnx_model.graph.value_info:
-            for dim_i in range(len(v.type.tensor_type.shape.dim)):
-                dim = v.type.tensor_type.shape.dim[dim_i]
+        inputs = list(onnx_model.graph.input)
+        outputs = list(onnx_model.graph.output)
+        valueinfos = list(onnx_model.graph.value_info)
+        for v in inputs + outputs + valueinfos:
+            for dim in v.type.tensor_type.shape.dim:
                 if dim.dim_param:
                     symbol_shape_set.add(dim.dim_param)
         return len(symbol_shape_set)
 
-    def test_clip_enable_symbolic(self):  # type: () -> None
+    def _get_shape_from_name(self, onnx_model, name):  # type: (ModelProto, Text) -> Optional[Sequence[Union[Text, int]]]
+        shape = []
+        inputs = list(onnx_model.graph.input)
+        outputs = list(onnx_model.graph.output)
+        valueinfos = list(onnx_model.graph.value_info)
+        for v in inputs + outputs + valueinfos:
+            if v.name == name:
+                for dim_i in range(len(v.type.tensor_type.shape.dim)):
```

Comment:
may be better to return ```v.type.tensor_type.shape``` here and have a separate equality-check method for two shapes, instead of creating another alternative shape representation. Also, would be useful to document assumptions for use of this method (e.g., whether the type is expected to be a tensor_type, or should we be checking that also here).

Reply:
Finally I chose to return shape without other check method and directly compare whether the TensorShapeProto is the same. Or I checked sparse_tensor_proto as well. Thanks!